src/main/java/com/sonyericsson/jenkins/plugins/bfa/sod/ScanOnDemandTask.java

```diff
@@ -66,14 +66,16 @@ public void run() {
                 for (AbstractBuild run : runs) {
                     if (run.getActions(FailureCauseBuildAction.class).isEmpty()
                             && run.getActions(FailureCauseMatrixBuildAction.class).isEmpty()
-                            && run.getResult().isWorseThan(Result.SUCCESS)
+                            && ResultFilter.needToAnalyze(run.getResult())
                             && run.getNumber() == build.getNumber()) {
                         scanBuild(run);
                     }
                 }
                 endMatrixBuildScan();
             } else {
-                scanBuild(build);
+                if (ResultFilter.needToAnalyze(build.getResult())) {
```

Comment:
This is for a manual scan and should not concern itself with whether it should scan it or not, the user has explicitly asked for the build to be scanned so it should be.